lib/scaffold/start.js

```diff
@@ -138,30 +138,39 @@ function cleanShutdown(_process, node) {
  * @param {Node} node
  */
 function registerExitHandlers(_process, node) {
-
-  function exitHandler(options, err) {
-    if (err) {
-      log.error('uncaught exception:', err);
-      if(err.stack) {
-        log.error(err.stack);
-      }
-      node.stop(function(err) {
-        if(err) {
-          log.error('Failed to stop services: ' + err);
-        }
-        _process.exit(-1);
-      });
-    }
-    if (options.sigint) {
-      cleanShutdown(_process, node);
-    }
-  }
-
   //catches uncaught exceptions
-  _process.on('uncaughtException', exitHandler.bind(null, {exit:true}));
+  _process.on('uncaughtException', exitHandler.bind(null, {exit:true}, _process, node));
 
   //catches ctrl+c event
-  _process.on('SIGINT', exitHandler.bind(null, {sigint:true}));
+  _process.on('SIGINT', exitHandler.bind(null, {sigint:true}, _process, node));
+}
+
+/**
+ * Will handle all the shutdown tasks that need to take place to ensure a safe exit
+ * @param {Object} options
+ * @param {String} options.sigint - The signal given was a SIGINT
+ * @param {Array}  options.exit - The signal given was an uncaughtException
+ * @param {Object} _process - The Node.js process
+ * @param {Node} node
+ * @param {Error} error
+*/
+function exitHandler(options, _process, node, err) {
+  if (err) {
+    log.error('uncaught exception:', err);
+    if(err.stack) {
+      log.error(err.stack);
+    }
+    node.stop(function(err) {
+      if(err) {
+        log.error('Failed to stop services: ' + err);
+      }
+      _process.exit(-1);
+    });
+  }
+  if (options.sigint) {
+    _process.on('SIGINT', function(){});
```

Comment:
This code isn't doing anything. Are you trying to remove the other event listener that was bound to SIGINT? For that you need to use `removeListener()`.